### ISSUE TITLE
Error when updating batch account after creation with terraform all version

### ISSUE THREAD

**Issue by MoussKone** [hashicorp/terraform-provider-azurerm#16660]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.9

### AzureRM Provider Version

3.4.0

### Affected Resource(s)/Data Source(s)

azurerm_batch_account

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "batch-rg" {
  name     = "testbatch"
  location = "West Europe"
}

resource "azurerm_batch_account" "batch" {
  name                 = "testbatchaccount"
  resource_group_name  = azurerm_resource_group.batch-rg.name
  location             = azurerm_resource_group.batch-rg.location
  pool_allocation_mode = "BatchService"

  tags = {
    env = "test"
    tobeDeletedForupdate = "DeleteThisTagAfter"
  }
}
    
terraform {
  required_providers {
    azurerm = {
      source = "hashicorp/azurerm"
      version = "3.4.0"
    }
  }
}

provider "azurerm" {
  features {}
}
```

### Debug Output/Panic Output

```shell
https://github.com/MoussKone/TerraformBatchError/blob/main/LOG.txt
```

### Expected Behaviour

we want to update our batch without needing to add a storage account to it because we are creating the batch without attaching a storage account to it by terraform

### Actual Behaviour

once the batch account is created without a storage account attached to it, we can no longer update the batch by terraform knowing that the storage_account_id argument of the azurerm_batch_account resource is optional with terraform according to the official documentation.

no matter what modification we want to make by terraform on the batch account we have this error: 

Error: updating Batch account "testbatchaccount" (Resource Group "testbatch"): batch.AccountClient#Update: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="LinkedInvalidPropertyId" Message="Property id '' at path 'properties.autoStorage.storageAccountId' is invalid. Expect fully qualified resource Id that start with '/subscriptions/{subscriptionId}' or '/providers/{resourceProviderNamespace}/'."

### Steps to Reproduce

1. copy My exemple configuration file terraform by creating a batch account without storage account.
2. make the terraform init then terraform plan finally terraform apply
3. just in the configuration try to delete a tag to update the batch account with terraform
4. make again the step number 3, you'll have the same issue

**Comment by sinbai** [comment]
@MoussKone thank you for opening this issue.  PR #16731  has been submitted to fix it. Please track it for more updates.

**Comment by sinbai** [comment]
@MoussKone since the issue has been released in [v3.12.0 of the Terraform Provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.12.0/CHANGELOG.md), are you okey if we close this issue?

**Comment by mybayern1974** [comment]
Per sinbai's suggestion I'm closing this issue. Feel free to call out if this issue needs to be re-active.
